### DISCUSSION TITLE
How can I model `GreaterEqual` and `LessEqual` as `LinearInequalityConstraint`

### DISCUSSION THREAD

**Discussion by axkr** [Hipparchus-Math/hipparchus#334] [Q&A]
As far as I understand the API, I can model the `Greater` and `Less` relations in a `LinearInequalityConstraint` for the `SQPOptimizerGM` optimizer.
@roccafrancesco, @Serrof: how can I model the `GreaterEqual` and `LessEqual` relations?

**Comment by roccafrancesco** [comment]
hi
inequality constraints should be understood as greaterequal.
I would like to remark that, at moment, SQPOptimizerGM work starting from feasibility solution. If you don't have good starting point i suggest to use  SQPOptimizerS.

**Comment by axkr** [comment]
@roccafrancesco 

> If you don't have good starting point i suggest to use `SQPOptimizerS`.

I've now implemented your suggestion here:
- https://github.com/axkr/symja_android_library/commit/87857a8d29b721cb9879c7188f46f0071e9962b3

The conversion of the **linear binary relations** is implemented here:
- https://github.com/axkr/symja_android_library/blob/ce7ae3cde2a7809372c00e11f9bae45d766f9bc2/symja_android_library/matheclipse-core/src/main/java/org/matheclipse/core/reflection/system/FindMinimum.java#L213

`LessEqual` and `Less` use the same conversions at the moment.
`GreaterEqual` and `Greater` use also the same conversions at the moment.

Maybe you can also suggest some typically "default values", which should be set in the constructor of  `SQPOptimizerS` ?
And if the user doesn't set a preferred method, which method should be used as default?
(Currently `"Powell"`, `"ConjugateGradient"` and `"SequentialQuadratic"` are available)


You can test [`FindMinimum`](https://github.com/axkr/symja_android_library/blob/master/symja_android_library/doc/functions/FindMinimum.md)  and [`FindMaximum`](https://github.com/axkr/symja_android_library/blob/master/symja_android_library/doc/functions/FindMaximum.md) in the demonstration web interface:
- https://matheclipse.org/input?i=FindMinimum%28%7Bx%2By%2C3%2Ax%2B2%2Ay%20%3E%3D%207%20%2C%20x%20%3E%3D%200%20%2C%20y%20%3E%3D%200%7D%2C%20%7Bx%2C%20y%7D%2CMethod%20-%3E%22SequentialQuadratic%22%29

**Comment by roccafrancesco** [comment]
class SQPOption contain already defaul values.  Anyway you could play with eps and convCriteria leaving unchenged the other parameters.
There is not a preferred method. Depened of the nature of the problem. 

In the example  https://matheclipse.org/i](https://matheclipse.org/input?i=FindMinimum%28%7Bx%2By%2C3%2Ax%2B2%2Ay%20%3E%3D%207%20%2C%20x%20%3E%3D%200%20%2C%20y%20%3E%3D%200%7D%2C%20%7Bx%2C%20y%7D%2CMethod%20-%3E%22SequentialQuadratic%22%29)
problem is linear and you could use a linear method  SimplexSolver.
